internal/discovery/foreign-cluster-operator/foreign-cluster-controller.go

```diff
@@ -508,8 +509,29 @@ func (r *ForeignClusterReconciler) getAddress() (string, error) {
 		}
 	}
 
+	// only physical nodes
+	//
+	// we need to get an address from a physical node, if we have established peerings in the past with other clusters,
+	// we may have some virtual nodes in our cluster. Since their IPs will not be reachable from other clusters, we cannot use them
+	// as address for a local NodePort Service
+	labelSelector, err := metav1.LabelSelectorAsSelector(&metav1.LabelSelector{
+		MatchExpressions: []metav1.LabelSelectorRequirement{
+			{
+				Key:      "type",
+				Operator: metav1.LabelSelectorOpNotIn,
+				Values:   []string{"virtual-node"},
```

Comment:
Just a minor comment, it would probably better to refer to constants instead of using hard-coded values, to avoid issues if the key or the value are changed in the future